### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by joszo**

### CODE REVIEW

File: src/create-project.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import * as latestVersion from 'latest-version';
+import * as shell from 'shelljs';
+import * as mkdirp from 'mkdirp';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import * as json from '../template/package.json';
```
Comment (tgorskievo) [inline src/create-project.ts]: I would rather change the name from `json` to e.g. `templatePackageJson` or something that inform what it is.

File: src/create-project.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+const createPackageJson = (directory) => {
+    const kakuninConfigPath = path.resolve(directory);
+  
+    mkdirp(kakuninConfigPath, function (err) {
```
Comment (tgorskievo) [inline src/create-project.ts]: I think you can change it to the arrow function.

File: template/package.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "kakunin": "cross-env NODE_ENV=prod kakunin"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "cross-env": "^5.2.0",
+    "kakunin": "next"
```
Comment (tgorskievo) [inline template/package.json]: next?

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "commonjs",
+    "allowSyntheticDefaultImports": false,
+    "resolveJsonModule": true,
+    "outDir": "dist",
+    "noUnusedLocals": true,
+    "noImplicitAny": false,
```
Comment (tgorskievo) [inline tsconfig.json]: We should keep this flag enabled,  to make sure that all types are set.

Comment (joszo) [reply tsconfig.json]: changed

File: template/package.json (hunk 3 above)
Comment (joszo) [reply template/package.json]: Changed into 2.5.0

File: src/create-project.ts (hunk 1 above)
Comment (joszo) [reply src/create-project.ts]: Changed into templatePackageJson

File: package.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    }
+  ],
+  "main": "./dist/index.js",
+  "bin": {
+    "kakunin-cli": "./dist/index.js"
+  },
+  "scripts": {
+    "build": "tsc"
+  },
+  "devDependencies": {
```
[the remark this answers is not in the document]
Comment (joszo) [reply package.json]: Moved necessary dependencies into 'dependencies' excluding types and typescript.

File: src/create-project.ts (hunk 2 above)
Comment (joszo) [reply src/create-project.ts]: Changed into => function.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
```
Comment (tgorskievo) [inline package.json]: Now you have duplicated dependencies

File: src/create-project.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import * as latestVersion from 'latest-version';
+import * as shell from 'shelljs';
+import * as mkdirp from 'mkdirp';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import * as templatePackageJson from '../template/package.json';
+
+const createPackageJson = (directory : any) => {
```
Comment (tgorskievo) [inline src/create-project.ts]: Avoid using `any` - this should be `string`

File: src/create-project.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+        shell.cd(kakuninConfigPath)
+        shell.exec('npm i && npm run kakunin init')
+  
+        console.log("Project created.");
+      }
+    });
+  };
+  
+export const initProject = (directory : any) => (ver : any) => {
```
Comment (tgorskievo) [inline src/create-project.ts]: Same as above - `(directory : string) => (ver : string) => {`

**Pull request closed without merging**